### ISSUE TITLE
Chat room skeleton

### ISSUE THREAD

**Issue by lawrence-forooghian** [ably-labs/ably-chat-swift#19]
Replacement for [ECO-4901](https://ably.atlassian.net/browse/ECO-4901) so that we can discuss / link in the open. Text of that issue:

> Awaiting on a spec from the Chat-JS team but current understanding:
> 
> The room lifecycle manager is the most complicated piece when accounting for the implicit attach/detach requirements. This task is responsible for making a simple attempt at attaching and detaching to rooms, without additional complexities uncovered in the spec. They will be tackled in a later task. 

[ECO-4901]: https://ably.atlassian.net/browse/ECO-4901?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4936) by [Unito](https://www.unito.io)

**Comment by lawrence-forooghian** [comment]
I misunderstood the scope of this task; I thought it was to implement Andy’s spec in https://github.com/ably/specification/pull/200/. That's not what it is. The idea is to implement the following:

- `Rooms.get` to return a room
- and then `Room.attach()` will then attempt to attach to all of the channels listed in https://ably.atlassian.net/wiki/spaces/CHA/pages/3083501593/CHADR-025+Chat+Permissions+and+Implicit+Permissions+for+Realtime+Channels, succeeding if and only if all of these calls succeed
- ditto for `Room.detach()`, detaching from the channels
- emit `ATTACHED` and `DETACHED` room status updates if `attach` / `detach` method succeeds, respectively

**Comment by lawrence-forooghian** [comment]
Also we won't implement this in the example app as part of this task; will revisit at start of next sprint where hopefully the skeleton of the example app will have been built out.
